components/javascript-wrapper.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * This file contacts the common code used by all of the variations of any RDF
+ * node componentin the noflo implementation of the RDF Pipeline.
+ */
+
+var _ = require('underscore');
+
+// TODO: Revisit with James' latest work
+var promiseComponent = require('./promise-component');
+var rpf = require('./rpf');
+
+/**
+ * This module provides a Javascript wrapper for updaters.  It is responsible for setting up the
+ * code to ensure updater callback, and then invoking the RDF pipeline framework to complete building
+ * the component and setting up the vni and state metadata.
+ *
+ * @param nodeDef updater node definition. This can be a simple updater function, or an object that contains 
+ *        updater metadata (e.g., input/output port definitions, description, icons), and an updater.
+ *
+ * @return a promise to create the noflo rdf component
```

Comment:
Why not create the noflo component directly, instead of returning a promise to do it?  I can see the need for Promises for asynchronous code that is executed at runtime (i.e., when the pipeline is running).  But he process of creating the components for the pipeline can be entirely synchronous, can't it?  Why are Promises needed for that?